akka-persistence/src/test/java/akka/persistence/fsm/AbstractPersistentFSMTest.java

```diff
@@ -129,13 +129,13 @@ public void fsmTimeoutTest() throws Exception {
             PersistentFSM.Transition stateTransition = expectMsgClass(PersistentFSM.Transition.class);
             assertTransition(stateTransition, fsmRef, UserState.LOOKING_AROUND, UserState.SHOPPING);
 
-            within(duration("0.9 seconds"), remainingOrDefault(), () -> {
+            within(java.time.Duration.ofMillis(900), getRemainingOrDefault(), () -> {
```

Comment:
Import the Java duration into scope instead, and use directly, if it clashes with the Scala one, use the FQCN for that instead

Reply:
Thanks for the reminder. I didn't import `java.time.Duration` because of this line:
https://github.com/akka/akka/blob/de97fbe8767f864d1d15afd8d7729433cda765ae/akka-persistence/src/test/java/akka/persistence/fsm/AbstractPersistentFSMTest.java#L514

If we want to remove `scala.concurrent.duration.Duration`, I need to add an overloaded method for this:   
https://github.com/akka/akka/blob/de97fbe8767f864d1d15afd8d7729433cda765ae/akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala#L359
However, I am not familiar enough with Akka Persistence, I don't know is it fine to add an overloaded method. 

If there is no any consideration, this is what I would add: 
``` scala
def forMax(timeout: java.time.Duration): State[S, D, E] = timeout match {
  case f: java.time.Duration ⇒ copy(timeout = Some(f.asScala))
  case _                     ⇒ copy(timeout = PersistentFSM.SomeMaxFiniteDuration)
}
```

Thanks! :)

Reply:
If you want to postpone the PersistentFSM change we can use the `scala.concurrent.duration.Duration` FQCN for just the `forMax` call? That's the 'odd one out' after all.

Reply:
@raboof Oh, you're right. Thanks! I will use FQCC for `forMax`.